src/extension/csv-with-metadata-header/src/main/java/au/org/emii/geoserver/wfs/response/CSVWithMetadataHeaderOutputFormat.java

```diff
@@ -26,51 +14,168 @@
         BufferedWriter w = new BufferedWriter(new OutputStreamWriter(output));
-        writeMetadata(featureCollection, getMetadataFeatureName(featureCollection), w);
-        this.csvOutputFormat.write(featureCollection, output, getFeature);
+        //writeMetadata(featureCollection, getMetadataFeatureName(featureCollection), w);
+        writeData(featureCollection, output, getFeature);
+    }
+
+
+    public void writeData(FeatureCollectionResponse featureCollection, OutputStream output, Operation getFeature) throws IOException, ServiceException {
+        BufferedWriter w = new BufferedWriter(new OutputStreamWriter(output, this.gs.getGlobal().getSettings().getCharset()));
+        FeatureCollection<?, ?> fc = (FeatureCollection)featureCollection.getFeature().get(0);
+        Object att;
+
+        SimpleFeatureType ft = (SimpleFeatureType)((FeatureCollection)fc).getSchema();
+        w.write("FID,");
+
+        for(int i = 0; i < ft.getAttributeCount(); ++i) {
+            AttributeDescriptor ad = ft.getDescriptor(i);
+            w.write(this.prepCSVField(ad.getLocalName()));
+            if (i < ft.getAttributeCount() - 1) {
+                w.write(",");
+            }
+        }
+
+        w.write("\r\n");
+        NumberFormat coordFormatter = NumberFormat.getInstance(Locale.US);
+        coordFormatter.setMaximumFractionDigits(this.getInfo().getGeoServer().getSettings().getNumDecimals());
+        coordFormatter.setGroupingUsed(false);
+        FeatureIterator i = ((FeatureCollection)fc).features();
+
+        try {
+            for(; i.hasNext(); w.write("\r\n")) {
+                Feature f = i.next();
+                w.write(this.prepCSVField(f.getIdentifier().getID()));
+                w.write(",");
+
+                for(int j = 0; j < ((SimpleFeature)f).getAttributeCount(); ++j) {
+                    att = ((SimpleFeature)f).getAttribute(j);
+                    if (att != null) {
+                        String value = formatToString(att, coordFormatter);
+                        w.write(this.prepCSVField(value));
+                    }
+
+                    if (j < ((SimpleFeature)f).getAttributeCount() - 1) {
+                        w.write(",");
+                    }
+                }
+            }
+        } finally {
+            i.close();
+        }
+        w.flush();
+    }
+
+    private String prepCSVField(String field) {
+        String mod = field.replaceAll("\"", "\"\"");
+        if (CSV_ESCAPES.matcher(mod).find()) {
+            mod = "\"" + mod + "\"";
+        }
+        return mod;
+    }
+
+    public String formatToString(Object att, NumberFormat coordFormatter) {
+        String value = null;
+
+        if (att instanceof Number) {
+            value = coordFormatter.format(att);
+        } else if (att instanceof Date) {
+            if (att instanceof java.sql.Date) {
+                value = utcDateString((Date)att);
```

Comment:
I'd leave this as serializeSqlDate.  By definition sql date is a date without time and without timezone so displaying it with a time and timezone doesn't sound right.  I don't think we'll ever use one of these - it sounds problematic.